MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -32,323 +29,273 @@ namespace MediaBrowser.MediaEncoding.Encoder
-            config.EncoderAppPath = path;
-            ConfigurationManager.SaveConfiguration("encoding", config);
-
-            Init();
-        }
-
-        private bool ValidateVersion(string path, bool logOutput)
-        {
-            return new EncoderValidator(_logger, _processFactory).ValidateVersion(path, logOutput);
-        }
-
-        private void ConfigureEncoderPaths()
-        {
-            if (_hasExternalEncoder)
-            {
-                return;
-            }
+                else
+                {
+                    // Supplied path could be either file path or folder path.
+                    // Resolve down to file path and validate
+                    path = GetEncoderPath(path);
 
-            var appPath = GetEncodingOptions().EncoderAppPath;
+                    if (path == null)
+                    {
+                        throw new ResourceNotFoundException("FFmpeg not found");
+                    }
+                    else if (!ValidatePath(FFtype.FFmpeg, path, "New From UI"))
+                    {
+                        throw new ResourceNotFoundException("Failed validation checks.  Version 4.0 or greater is required");
+                    }
+                    else
+                    {
+                        FFmpegPath = path;
 
-            if (string.IsNullOrWhiteSpace(appPath))
-            {
-                appPath = Path.Combine(ConfigurationManager.ApplicationPaths.ProgramDataPath, "ffmpeg");
-            }
+                        // Write the validated mpeg path to the xml as <EncoderAppPathCustom>
+                        // This ensures its not lost on new startup
+                        var config = ConfigurationManager.GetConfiguration<EncodingOptions>("encoding");
+                        config.EncoderAppPathCustom = FFmpegPath;
+                        ConfigurationManager.SaveConfiguration("encoding", config);
 
-            var newPaths = GetEncoderPaths(appPath);
-            if (string.IsNullOrWhiteSpace(newPaths.Item1) || string.IsNullOrWhiteSpace(newPaths.Item2) || IsSystemInstalledPath(appPath))
-            {
-                newPaths = TestForInstalledVersions();
-            }
+                        FFprobePath = null; // Clear probe path so it gets relooked in ReInit()
 
-            if (!string.IsNullOrWhiteSpace(newPaths.Item1) && !string.IsNullOrWhiteSpace(newPaths.Item2))
-            {
-                FFMpegPath = newPaths.Item1;
-                FFProbePath = newPaths.Item2;
+                        ReInit();
+                    }
+                }
             }
-
-            LogPaths();
         }
 
-        private Tuple<string, string> GetEncoderPaths(string configuredPath)
+        private bool ValidatePath(FFtype type, string path, string logComment)
         {
-            var appPath = configuredPath;
-
-            if (!string.IsNullOrWhiteSpace(appPath))
+            if (!string.IsNullOrEmpty(path))
```

Comment:
File.Exists returns false if path is null or empty so this check is unnecessary

Reply:
Agreed File.Exists() is null safe, but only want to print warnings to Log file if a path was given.